lib/AL_USDMaya/AL/usdmaya/Global.cpp

```diff
@@ -98,6 +99,56 @@ AL::event::CallbackId Global::m_preRead;
 AL::event::CallbackId Global::m_postRead;
 AL::event::CallbackId Global::m_fileNew;
 
+//----------------------------------------------------------------------------------------------------------------------
+//class of MObjects
+MSelectionList selected;
+
+//Store the current selection list, but dont store AL_USD proxies
+static void storeSelection()
+{
+  MGlobal::displayInfo("storeSelection()");
+  //set "selected" to the current selection list
+  MGlobal::getActiveSelectionList(selected);
+  for( int i=0; i<selected.length(); ++i )
+  {
+    MObject obj;
+    selected.getDependNode(i,obj);
+    MFnDependencyNode fnParent(obj);
+    // Remove if type is AL_usdmaya_Transform
+    if (fnParent.typeName() == "AL_usdmaya_Transform")
+    {
+      MGlobal::unselectByName(fnParent.name().asChar());
+    }
+    MFnDagNode fnDagNode(obj);
+    // Unselect nodes which have AL_usdmaya_ProxyShape as a child
+    for( int i=0; i!=fnDagNode.childCount(); ++i ) {
+      MObject obj = fnDagNode.child(i);
+      MFnDagNode fnChild(obj);
+      if (fnChild.typeName() == "AL_usdmaya_ProxyShape")
+      {
+        MGlobal::unselectByName(fnParent.name().asChar());
+        MGlobal::unselectByName(fnChild.name().asChar());
+      }
+    }
```

Comment:
I'm not sure the logic here is correct? The following would pass this logic and leave the selection untouched:

```select -r `ls -type "AL_usdmaya_ProxyShape"` ``` 

This should do more or less the same thing, but with a slight lean towards a more efficient approach. Only issue I can see with this code, is if a parent->parent->proxyshape is selected (since neither approach catches that case? is that important?)

```c++

// some utils that test for AL types, but which only initialise function sets when it's possible
// that the type may be a plugin shape or transform. Avoids a tonne of function set initialisations
// and string compares on the types. 
auto isProxyShape = [] (MDagPath p) {
  if(p.node().hasFn(MFn::kPluginShape)) {
     return MFnDagNode(p).typeName() == "AL_usdmaya_ProxyShape";
  }
  return false;
};
auto isTransform = [] (MDagPath p) {
  if(p.node().hasFn(MFn::kPluginTransform)) {
     return MFnDagNode(p).typeName() == "AL_usdmaya_Transform";
  }
  return false;
};

for(int i = 0; i < selected.length(); /* empty */ )
{
  // grab item as a dag path (skip over materials/textures/etc)
  MDagPath selectedPath; 
  if(!selection.getDagPath(i, selectedPath))
  {
    ++i;
    continue;
  }

  // test for any selected proxy shapes or transform nodes
  if(isProxyShape(selectedPath) || isTransform(selectedPath))
  {
    // remove node from selection list
    selection.remove(i);
    continue;
  }

  // test for any parents of proxy shapes selected (don't iterate over all children, just the shape nodes below)
  uint32_t num = 0, j;
  selectedPath.numberOfShapesDirectlyBelow(num);
  for(j = 0; j < num; ++j)
  {
    MDagPath child = selectedPath;
    child.extendToShapeDirectlyBelow(j); //< only care about shape nodes (rather than ALL children!)
    if(isProxyShape(child))
    {
      selection.remove(i);
      removed = true;
      break;
    }
  }

  // if none found, increment count
  if(j == num)
  {
    ++i;
  }
}
```